log/logger.go

```diff
@@ -1,33 +1,61 @@
 package log
 
 import (
-	"fmt"
+	"context"
 	"os"
+	"runtime"
 	"time"
 
-	"github.com/go-stack/stack"
+	"math"
+	"golang.org/x/exp/slog"
 )
 
-const timeKey = "t"
-const lvlKey = "lvl"
-const msgKey = "msg"
-const ctxKey = "ctx"
-const errorKey = "LOG15_ERROR"
-const skipLevel = 2
+// TODO: define custom trace level and pull in whatever else from slog
 
-type Lvl int
+const (
+	timeKey = "t"
+	lvlKey  = "lvl"
+	msgKey  = "msg"
+	// TODO: change the below
+	errorKey = "LOG15_ERROR"
+
+	// TODO: not really clear what the use of the context key was in log15
+	// the context should encompass all values that are to be printed
+	// so having a separate key for context doesn't make sense to me
+	//ctxKey = "ctx"
+
+	// TODO: re-incorporate this here instead of inside func
+)
 
+// we use custom error levels here (instead of default provided from slog) to keep backwards
+// compatibility with pre-existing geth log lib
```

Comment:
Just, in general, how much 'cruft' could we get rid of if we didn't care a whole lot about backwards compatibility? What's your general feeling?

Reply:
This particular detail is going away in the refactor:

There will be no notion of a separate `Level` in the Geth `log` package.  We will convert internally to match the level values used by slog (e.g. in `Vmodule()`).

Reply:
Other than that, I'm not sure. I thought a bit about using the built-in handlers (json and text handlers).  But the text handler seems basic (only supports logfmt, not terminal format).

One issue is that these are standalone handlers and not "formatters" (which don't exist in `slog`).  So I would lean to keep the existing formatters because 1) we know they work and 2) we already have a distinction between Handler and Formatter which is kind of nice IMO.

Reply:
also, because `slog` uses `Record.PC` instead of associating a call stack to the record, I remove the reliance on `go-stack/stack`.  The only downside is that this has some interaction with the `LazyHandler` where it is harder to extract the relevant portion of a call-stack if one is returned from a lazy evaluation (we can't use the utilities provided by `go-stack/stack` to trim [here](https://github.com/ethereum/go-ethereum/blob/4e1e37323d87967708b58546f040990db7c2820c/log/handler.go#L285-L288) if we don't have a `CallStack` object associated with the `Record`, available to use).

Since we don't use this functionality anywhere, I just remove this trimming.

Reply:
Also I assume that we don't want to keep any previously-built-in handler that we aren't using in the Geth codebase.  So I've removed them.  Users can just re-implement on their own.